api/v1alpha1/openapi.yaml

```diff
@@ -2502,6 +2511,254 @@ components:
           type: string
           description: 'Status of the operation. One of: "Success" or "Failure". More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status'
       description: Status is a return value for calls that don't return other objects.
+    DeviceWorkloadStatus:
+      type: object
+      required:
+        - summary
+      properties:
+        applications:
+          type: array
+          description: 'Indicates the current status of applications running on the device.'
+          items:
+            $ref: '#/components/schemas/DeviceApplicationStatus'
+        summary:
+          $ref: '#/components/schemas/DeviceWorkloadSummaryStatus'
+          description: 'Summary status of device workloads.'
+    DeviceApplicationStatus:
+      type: object
+      required:
+        - id
+        - name
+        - status
+      properties:
+        id:
```

Comment:
True but for other resources the name is generated by the service. And even if the service did create this name it would not be a unique instance. For example, a device reboots in terms of auditing when it starts is a second instance of an application. Also if the service or user defines a pod, when it starts that name will be different from the declared name.

It felt like it was needed to track lifecycle phases and would not necessarily be needed to be pushed because its optional. 

I guess I can wrap status in another object during implementation and we can drop it here.

Reply:
removed